lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -285,11 +392,119 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
+        return (
+            (this.editable && this.hasMetadataChanged) ||
+            (this.editableTags && this.hasMetadataChanged) ||
+            (this.editableCategories && this.hasMetadataChanged) ||
+            (this.editableGroup?.editable && this.hasMetadataChanged)
+        );
+    }
+
+    toggleEdit(event: MouseEvent, group: CardViewGroup, buttonType: ButtonType): void {
+        event.stopPropagation();
+        if (this.isEditingPanel()) {
+            this.notificationService.showError('METADATA.BASIC.SAVE_OR_DISCARD_CHANGES');
+            return;
+        }
+
+        if (this.editableGroup && this.editableGroup !== group) {
+            this.editableGroup.editable = false;
+        }
+
+        switch (buttonType) {
+            case ButtonType.GeneralInfo:
+                this.editable = !this.editable;
+                this.editableChange.emit(this.editable);
+                this.panel.open();
+                group.editable = false;
+                break;
+            case ButtonType.Tags:
+                this.editableTags = !this.editableTags;
+                this.editableTagsChange.emit(this.editableTags);
+                this.isTagPanelVisible = this.editableTags;
+                this.tagNameControlVisible = true;
+                group.editable = false;
+                break;
+            case ButtonType.Categories:
+                this.editableCategories = !this.editableCategories;
+                this.editableCategoriesChange.emit(this.editableCategories);
+                this.isCategoriesPanelVisible = this.editableCategories;
+                this.categoryControlVisible = true;
+                group.editable = false;
+                break;
+            case ButtonType.Group:
+                group.editable = !group.editable;
+                this.groupChange.emit(group);
+                this.editableGroup = group.editable ? group : null;
+                if (group.editable) {
+                    group.expanded = true;
+                }
+                break;
+            default:
+                break;
+        }
+
+        if (buttonType !== ButtonType.GeneralInfo) {
+            this.editable = false;
+        }
+
+        if (buttonType !== ButtonType.Tags) {
+            this.editableTags = false;
+        }
+
+        if (buttonType !== ButtonType.Categories) {
+            this.editableCategories = false;
+        }
+    }
+
+    toggleGeneralInfoPanel() {
+        this.isGeneralInfoPanelVisible = !this.isGeneralInfoPanelVisible;
+        this.cdr.detectChanges();
+    }
+
+    toggleTagsPanel(tagPanelState: boolean) {
+        this.isTagPanelVisible = tagPanelState;
+        this.cdr.detectChanges();
+    }
+
+    toggleCategoriesPanel(isCategoriesPanelVisible: boolean) {
+        this.isCategoriesPanelVisible = isCategoriesPanelVisible;
+        this.cdr.detectChanges();
+    }
+
+    isTagsEmpty(): boolean {
+        return this.tags?.length === 0 && !this.editableTags;
```

Comment:
Why do you check if tags are not editable inside `isTagsEmpty` method?

Reply:
We are showing the empty tag message on view mode. If we remove the condition then it will display on edit mode too. That is why we need the condition here.